### ISSUE TITLE
Rate limiting?

### ISSUE THREAD

**Issue by scalp42** [gmr/env-aws-params#13]
Hi @gmr,

Thanks for the tool. What's your take with rate limiting? Currently if you deploy multiple say 100 containers, you're going to run into rate limiting on AWS side.

Any chance to be able to handle say sleeping/retrying with maybe a flag instead of failing?

Cheers 🙌

**Comment by gmr** [comment]
Yes — since the migration to `aws-sdk-go-v2` (cf9246f), retries with exponential backoff are enabled by default:

- Default retry mode: `standard`
- Default max attempts: 3 (2 retries after the initial call)
- Throttling errors (`ThrottlingException`, HTTP 429) are retryable

For the 100-container fan-out scenario, you can tune this without code changes via the SDK's standard environment variables:

```
AWS_RETRY_MODE=adaptive    # token-bucket throttling, much better for fan-out
AWS_MAX_ATTEMPTS=10
```

`adaptive` mode is specifically designed for this case — it learns the throttling rate and slows the client down before SSM rejects requests.
